### ISSUE TITLE
Error: Something went wrong. cron reached maximum iterations

### ISSUE THREAD

**Issue by eimajenthat** [n8n-io/n8n#2239]
**Describe the bug**
I have a workflow that is triggered by a cron trigger node, configured to run weekly on Monday mornings.  It fails to run pretty consistently.  If I am logged in and using the n8n UI, the cron job seems to work.  It's only if I leave it alone that I have an issue.

When it fails, I do not see a failed execution in the executions list in the UI, as I do with other types of failures.  However, I do see an error in the docker logs if I run `docker logs`:

```
n8n_1  | Error: Something went wrong. cron reached maximum iterations.
n8n_1  |                                                Please open an  issue (https://github.com/kelektiv/node-cron/issues/new) and provide the following string
n8n_1  |                                                Time Zone: "" - Cron String: 44 1 10 * * 1 - UTC offset: -05:00 - current Date: Mon Sep 20 2021 15:01:57 GMT+0000
n8n_1  |     at CronTime._getNextDateFrom (/usr/local/lib/node_modules/n8n/node_modules/cron/lib/cron.js:235:12)
n8n_1  |     at CronTime.sendAt (/usr/local/lib/node_modules/n8n/node_modules/cron/lib/cron.js:156:17)
n8n_1  |     at CronTime.getTimeout (/usr/local/lib/node_modules/n8n/node_modules/cron/lib/cron.js:175:29)
n8n_1  |     at CronJob.start (/usr/local/lib/node_modules/n8n/node_modules/cron/lib/cron.js:613:31)
n8n_1  |     at Timeout.callbackWrapper [as _onTimeout] (/usr/local/lib/node_modules/n8n/node_modules/cron/lib/cron.js:665:29)
n8n_1  |     at listOnTimeout (internal/timers.js:554:17)
n8n_1  |     at processTimers (internal/timers.js:497:7)
```

The error seems to indicate that there is an issue with the node-cron package, and there may well be.  If you search for "cron reached maximum iterations" in the node-cron issue tracker, you get 31 open issues and 15 closed:

https://github.com/kelektiv/node-cron/issues?q=cron+reached+maximum+iterations

The vast majority of those issues are unhelpful.  A few mention an issue with Daylight Saving Time which may or may not have been resolved at this point.  However, there is no Daylight Saving Time correction involved in my scenario.  Some others mention that this error seems to crop up when there is elevated CPU usage.  So far as I can tell, nothing is causing high CPU usage for me.  Since the error mentions maximum iterations and I am only trying to execute one workflow at the time when this is running, I am wondering if there is an infinite loop being generated somewhere.

If this node-cron library is indeed broken.  Perhaps it would make sense to look for an alternative?

For my part, I'm considering generating a webhook for my script, and then adding a cron job to my host machine that executes a curl command on the webhook.  I think that will probably work, but it seems pretty dumb.  I'm really hoping there's a better solution out there.

I think the error is also causing the container to restart.  The next thing that appears in the logs is:

```
n8n_1  | n8n now checks for new versions and security updates. You can turn this off using the environment variable N8N_VERSION_NOTIFICATIONS_ENABLED to "false"
n8n_1  | For more information, please refer to https://docs.n8n.io/getting-started/installation/advanced/configuration.html
n8n_1  |
n8n_1  | n8n ready on 0.0.0.0, port 5678
n8n_1  | Version: 0.133.0
n8n_1  |  ================================
n8n_1  |    Start Active Workflows:
n8n_1  |  ================================
n8n_1  |    - Payroll - Start Payments
n8n_1  |      => Started
n8n_1  |    - Payroll - Send Notification
n8n_1  |      => Started
n8n_1  |
n8n_1  | Editor is now accessible via:
```

The log doesn't have timestamps and I'm checking after the fact, but that's my best guess.

**To Reproduce**
1. Create a workflow to be executed by a cron trigger.
2. The issue doesn't seem to want to happen while you are watching, so maybe schedule it to run a few hours in the future, and then repeat daily after that.  My cron job was weekly, but for troubleshooting, I would imagine you want it to run more frequently.
3. Wait until the scheduled execution time elapses.
4. Log in to n8n and see if the workflow executed.
5. Run `docker logs` on your n8n Docker container to see the error message.

**Expected behavior**
I would like my cron-triggered workflows to be executed reliably at the time specified.  And if the workflow fails to be executed, I would like an error added to the execution log in UI.

**Environment (please complete the following information):**
 - Host OS: Ubuntu Linux 20.04
 - Docker Container OS: Alpine Linux 3.11.7
 - n8n Version: 0.133.0
 - Node.js Version: v14.15.5
 - Database system: SQLite
 - Operation mode: I don't know what this means.  But I used the standard Docker install instructions in the docs, and tried not to screw with anything.  So I'm guessing it's the default, `own`.

**Additional context**
I am running n8n in a Docker container built using Docker-Compose.  I have included it here:

```
version: "3"

services:
  n8n:
    image: n8nio/n8n
    restart: always
    ports:
      - "5678:5678"
    environment:
      - N8N_BASIC_AUTH_ACTIVE=true
      - N8N_BASIC_AUTH_USER
      - N8N_BASIC_AUTH_PASSWORD
      - N8N_HOST=${SUBDOMAIN}.${DOMAIN_NAME}
      - N8N_PORT=5678
      - N8N_PROTOCOL=http
      - NODE_ENV=production
      - WEBHOOK_TUNNEL_URL=https://${SUBDOMAIN}.${DOMAIN_NAME}/
      - GENERIC_TIMEZONE=${GENERIC_TIMEZONE}
    volumes:
      - ${DATA_FOLDER}:/home/node/.n8n
      - ./local-files:/files
```

**Comment by ghassanmas** [comment]
I have dig into the problem [here](https://github.com/kelektiv/node-cron/issues/478), and as it seems, the new cron version 1.8  is mostly responsible for this bug. So the most current solution is to force corn to use version 1.7, here is an example [commit](https://github.com/geut/permanent-seeder/commit/f4a614cfbca4360dcdce53ee0235c82646f53542) to resolve, that I think also apply to n8n. 

### Check which corn version you are using
However to ensure that what I am saying is relevant, or that you are using the buggy version of cron just run this command in the _docker enviroment_ `npm list corn` if the output is: 

`cron@1.8.2 /usr/local/lib/node_modules/n8n/node_modules/cron
`
### Possible temporary fix 
Then that means is relevant, you can also try running the following to downgrade the corn (in the docker env): 
```
cd /usr/local/lib/node_modules/n8n
npm i cron@1.7.2
```

**Comment by eimajenthat** [comment]
Thanks for the quick response!  I logged in to my docker container, and then `cd`ed to that n8n node_modules directory (/usr/local/lib/node_modules/n8n), and here's the output I got:

```
/usr/local/lib/node_modules/n8n # npm list cron
n8n@0.133.0 /usr/local/lib/node_modules/n8n
+-- n8n-core@0.78.0
| `-- cron@1.8.2
`-- n8n-nodes-base@0.130.0
  `-- cron@1.8.2  deduped
```

So it does indeed seem that I am running version 1.8.2.  I will try the temporary fix and see if it works for me.  Thanks again!

**Comment by eimajenthat** [comment]
So far so good.  I generated a simple workflow to send me an email based on a cron trigger and it worked.  Will continue to monitor it.

**Comment by janober** [comment]
Thanks a lot. We did now set version 1.7.x. Will be released with the next version.

**Comment by janober** [comment]
Got released with n8n@0.144.0
